### PR TITLE
Fix large direct response

### PR DESCRIPTION
**Pull request by howardjohn**
This causes a NACK in Envoy

Fixes https://github.com/istio/istio/issues/42510

**Please provide a description of this PR:**

### CODE REVIEW

**Review by GregHanson — APPROVED** [review]

**Comment by GregHanson** [comment]
@howardjohn is release-1.16 impacted by this too?

**Comment by howardjohn** [comment]
afaik all versions, but I didn't test or anything like that

File: pilot/pkg/networking/core/v1alpha3/httproute.go [hunk 1]
```diff
@@ -207,6 +208,8 @@ func (configgen *ConfigGeneratorImpl) buildSidecarOutboundHTTPRouteConfig(
 		Name:             routeName,
 		VirtualHosts:     virtualHosts,
 		ValidateClusters: proto.BoolFalse,
+		// Control plane validates 1mb max size. Set this on data plane too to increase from default of 4k
```
Comment (hzxuzhonghu) [inline pilot/pkg/networking/core/v1alpha3/httproute.go:211]: nit: what do you mean by this `Control plane validates 1mb max size.`

Comment (howardjohn) [reply pilot/pkg/networking/core/v1alpha3/httproute.go:211]: istiod validation webhook

**Review by hzxuzhonghu — APPROVED** [review]

**Pull request merged**